# **DESAFÍO Nº2: ESTADÍSTICAS DESCRIPTIVAS Y PROBABILIDADES (PARTE II)**
### **Análisis Estadístico con Python**
#### *Carrera Data Science - G-93 - Desafio Latam*
#### Estudiante: Felipe Leal Arancibia

En este desafío validaremos nuestros conocimientos relacionados con probabilidades. Para lograrlo, necesitarás aplicar lo que has aprendido durante esa sesión.

Lee todo el documento antes de comenzar el desarrollo individual, para asegurarte de tener el máximo de puntaje y enfocar bien los esfuerzos.

Tiempo asociado: 2 horas cronológicas

## Descripción

#### Una casa de apuestas quiere perfeccionar sus modelos de estimaciones de factores de apuestas, por lo que quiere hacer una revisión sobre los resultados del Mundial 2014.

#### Para resolver este desafío deberás cargar datos desde el dataset worldcup2014.csv, y realizar las operaciones necesarias para determinar lo pedido.

In [1]:
# Se importan librerías para explorar bbdd en cvs
import pandas as pd
import numpy as np

In [2]:
# Se cargan los datos de archivo

mundial_2014 = pd.read_csv('worldcup2014.csv', delimiter=',')
mundial_2014

team     continent group  group_pos  cantidad_juegos  \
0                   Brazil  southamerica     A          1                3   
1                   Mexico  northamerica     A          2                3   
2                  Croatia        europe     A          3                3   
3                 Cameroon        africa     A          4                3   
4              Netherlands        europe     B          1                3   
5                    Chile  southamerica     B          2                3   
6                    Spain        europe     B          3                3   
7                Australia          asia     B          4                3   
8                 Colombia  southamerica     C          1                3   
9                   Greece        europe     C          2                3   
10             Ivory Coast        africa     C          3                3   
11                   Japan          asia     C          4                3   
12              Costa Rica  northamerica     D          1                3   
13                 Uruguay  southamerica     D          2                3   
14                   Italy        europe     D          3                3   
15                 England        europe     D          4                3   
16                  France        europe     E          1                3   
17             Switzerland        europe     E          2                3   
18                 Ecuador  southamerica     E          3                3   
19                Honduras  northamerica     E          4                3   
20               Argentina  southamerica     F          1                3   
21                 Nigeria        africa     F          2                3   
22  Bosnia and Herzegovina        europe     F          3                3   
23                    Iran          asia     F          4                3   
24                 Germany        europe     G          1                3   
25           United States  northamerica     G          2                3   
26                Portugal        europe     G          3                3   
27                   Ghana        africa     G          4                3   
28                 Belgium        europe     H          1                3   
29                 Algeria        africa     H          2                3   
30                  Russia        europe     H          3                3   
31             South Korea          asia     H          4                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0                2                 1                0            7   
1                2                 1                0            4   
2                1                 0                2            6   
3                0                 0                3            1   
4                3                 0                0           10   
5                2                 0                1            5   
6                1                 0                2            4   
7                0                 0                3            3   
8                3                 0                0            9   
9                1                 1                1            2   
10               1                 0                2            4   
11               0                 1                2            2   
12               2                 1                0            4   
13               2                 0                1            4   
14               1                 0                2            2   
15               0                 1                2            2   
16               2                 1                0            8   
17               2                 0                1            7   
18               1                 1                1            3   
19               0                 0           

#### 1. Considerando el experimento de seleccionar un país participante al azar, se pide determinar las probabilidades de los siguientes sucesos:

#### *Hint: define previamente condiciones para realizar los conteos*


In [4]:
# Se exploran y contabilizan los valores reiterados de la columna "continente"
mundial_2014["continent"].value_counts()

continent
europe          13
southamerica     6
africa           5
northamerica     4
asia             4
Name: count, dtype: int64

In [6]:
# Definición de condiciones por continente, pero con un filtro amplio, no sólo para booleanos de columna continente
equipos_europa = mundial_2014[mundial_2014["continent"]  == 'europe']
equipos_sudamerica = mundial_2014[mundial_2014["continent"]  == 'southamerica']
equipos_norteamerica = mundial_2014[mundial_2014["continent"]  == 'northamerica']
equipos_asia = mundial_2014[mundial_2014["continent"]  == 'asia']
equipos_africa = mundial_2014[mundial_2014["continent"]  == 'africa']

In [7]:
# Ejemplo
equipos_africa

team continent group  group_pos  cantidad_juegos  juegos_ganados  \
3      Cameroon    africa     A          4                3               0   
10  Ivory Coast    africa     C          3                3               1   
21      Nigeria    africa     F          2                3               1   
27        Ghana    africa     G          4                3               0   
29      Algeria    africa     H          2                3               1   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
3                  0                3            1             9   
10                 0                2            4             5   
21                 1                1            3             3   
27                 1                2            4             6   
29                 1                1            6             5   

   goles_diferencia  puntos  clasificado  
3                −8       0            0  
10               −1       3            0  
21                0       4            1  
27               −2       1            0  
29                1       4            1

In [9]:
equipos_asia

team continent group  group_pos  cantidad_juegos  juegos_ganados  \
7     Australia      asia     B          4                3               0   
11        Japan      asia     C          4                3               0   
23         Iran      asia     F          4                3               0   
31  South Korea      asia     H          4                3               0   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
7                  0                3            3             9   
11                 1                2            2             6   
23                 1                2            1             4   
31                 1                2            3             6   

   goles_diferencia  puntos  clasificado  
7                −6       0            0  
11               −4       1            0  
23               −3       1            0  
31               −3       1            0

In [10]:
# Se define un diccionario para mayor claridad de un próximo print de exploración inicial
dict_eq_continentes = {
    "Europa": equipos_europa, 
    "Sudamérica": equipos_sudamerica, 
    "África": equipos_africa, 
    "Norteamérica": equipos_norteamerica, 
    "Asia": equipos_asia} 


In [12]:
# Ciclo para imprimir los valores del diccionario (cantidad de equipos participantes por continente).
total_paises = 0
for continente, cantidad in dict_eq_continentes.items():
    cantidad_equipos = cantidad.shape[0]
    print(f"De {continente} participan {cantidad_equipos} selecciones.")
    total_paises += cantidad_equipos
print(f"Total cantidad de países participantes: {total_paises}" )

De Europa participan 13 selecciones.
De Sudamérica participan 6 selecciones.
De África participan 5 selecciones.
De Norteamérica participan 4 selecciones.
De Asia participan 4 selecciones.
Total cantidad de países participantes: 32


In [13]:
# En caso de querer ver la cantidad de equipos de un sólo continente se utilizará ".shape()"
#Ejemplo
equipos_sudamerica.shape[0]

6

In [14]:
# Crearé variables para facilitar este conteo por continentes

conteo_europa = equipos_europa.shape[0]
conteo_sudamerica = equipos_sudamerica.shape[0]
conteo_africa = equipos_africa.shape[0]
conteo_norteamerica = equipos_norteamerica.shape[0]
conteo_asia = equipos_asia.shape[0]


##### **a. pertenecer a África**

In [15]:
# probabilidad --> p = casos / total

print(f"Conteo selecciones Africa en el Mundial 2014: {conteo_africa}")
p_africa = round((conteo_africa / len(mundial_2014)),2)
f'La probabilidad de que una selección sea de África es de: {p_africa}'

Conteo selecciones Africa en el Mundial 2014: 5


'La probabilidad de que una selección sea de África es de: 0.16'

##### **b. pertenecer a Asia**

In [16]:
print(f"Conteo selecciones Asia en el Mundial 2014: {conteo_asia}")


p_asia = round((conteo_asia / len(mundial_2014)),2)
f'La probabilidad de que una selección sea de Asia es de: {p_asia}'

Conteo selecciones Asia en el Mundial 2014: 4


'La probabilidad de que una selección sea de Asia es de: 0.12'

##### **c. pertenecer a Europa**

In [17]:
print(f"Conteo selecciones Europa en el Mundial 2014: {conteo_europa}")


p_europa = round((conteo_europa / len(mundial_2014)),2)
f'La probabilidad de que una selección sea de Europa es de: {p_europa}'

Conteo selecciones Europa en el Mundial 2014: 13


'La probabilidad de que una selección sea de Europa es de: 0.41'

##### **d. pertenecer a Norteamérica**

In [18]:
print(f"Conteo selecciones Norteamérica en el Mundial 2014: {conteo_norteamerica}")



Conteo selecciones Norteamérica en el Mundial 2014: 4


In [19]:
# ¿Cuáles son? Porque tengo entendido que en América del Norte hay 3 países XD

equipos_norteamerica

team     continent group  group_pos  cantidad_juegos  \
1          Mexico  northamerica     A          2                3   
12     Costa Rica  northamerica     D          1                3   
19       Honduras  northamerica     E          4                3   
25  United States  northamerica     G          2                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
1                2                 1                0            4   
12               2                 1                0            4   
19               0                 0                3            1   
25               1                 1                1            4   

    goles_contra goles_diferencia  puntos  clasificado  
1              1                3       7            1  
12             1                3       7            1  
19             8               −7       0            0  
25             4                0       4            1

In [20]:
# Aah, incluyen Costa Rica y Honduras XD

In [21]:
p_norteamerica = round((conteo_norteamerica / len(mundial_2014)),2)
f'La probabilidad de que una selección sea de Norteamérica es de: {p_norteamerica}'

'La probabilidad de que una selección sea de Norteamérica es de: 0.12'

##### **e. pertenecer a Sudamérica**

In [22]:
print(f"Conteo selecciones Sudamérica en el Mundial 2014: {conteo_sudamerica}")


Conteo selecciones Sudamérica en el Mundial 2014: 6


In [23]:
p_sudamerica = round((conteo_sudamerica / len(mundial_2014)),2)
f'La probabilidad de que una selección sea de Sudamérica es de: {p_sudamerica}'

'La probabilidad de que una selección sea de Sudamérica es de: 0.19'

##### **f. clasificar a segunda ronda**

In [24]:
# Se considera que el valor 1 es "clasificados", por lo que se filtra el dataframe y se almacena en variable para futuras consultas
filtro_clasificados = mundial_2014[mundial_2014['clasificado']== 1]
filtro_clasificados

team     continent group  group_pos  cantidad_juegos  \
0          Brazil  southamerica     A          1                3   
1          Mexico  northamerica     A          2                3   
4     Netherlands        europe     B          1                3   
5           Chile  southamerica     B          2                3   
8        Colombia  southamerica     C          1                3   
9          Greece        europe     C          2                3   
12     Costa Rica  northamerica     D          1                3   
13        Uruguay  southamerica     D          2                3   
16         France        europe     E          1                3   
17    Switzerland        europe     E          2                3   
20      Argentina  southamerica     F          1                3   
21        Nigeria        africa     F          2                3   
24        Germany        europe     G          1                3   
25  United States  northamerica     G          2                3   
28        Belgium        europe     H          1                3   
29        Algeria        africa     H          2                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0                2                 1                0            7   
1                2                 1                0            4   
4                3                 0                0           10   
5                2                 0                1            5   
8                3                 0                0            9   
9                1                 1                1            2   
12               2                 1                0            4   
13               2                 0                1            4   
16               2                 1                0            8   
17               2                 0                1            7   
20               3                 0                0            6   
21               1                 1                1            3   
24               2                 1                0            7   
25               1                 1                1            4   
28               3                 0                0            4   
29               1                 1                1            6   

    goles_contra goles_diferencia  puntos  clasificado  
0              2                5       7            1  
1              1                3       7            1  
4              3                7       9            1  
5              3                2       6            1  
8              2                7       9            1  
9              4               −2       4            1  
12             1                3       7            1  
13             4                0       6            1  
16             2                6       7            1  
17             6                1       6            1  
20             3                3       9            1  
21             3                0       4            1  
24             2                5       7            1  
25             4                0       4            1  
28             1                3       9            1  
29             5                1       4            1

In [25]:
# Se almacena en variable el número de selecciones clasificadas
conteo_clasificados = filtro_clasificados.shape[0]
print(f"De las selecciones participantes, ****{conteo_clasificados}**** clasificaron a segunda ronda.") 


De las selecciones participantes, ****16**** clasificaron a segunda ronda.


In [26]:

# prob = casos / total
# Calcular la probabilidad de pasar a segunda ronda
p_clasificados = round((conteo_clasificados / len(mundial_2014)),2)
print(f'La probabilidad teória de que una selección clasifique a segunda ronda es:******* {p_clasificados} *******, sin atender experiencia previa u otros antecedentes.')



La probabilidad teória de que una selección clasifique a segunda ronda es:******* 0.5 *******, sin atender experiencia previa u otros antecedentes.


In [27]:
#La lista de selecciones que clasificaron a segunda ronda es la siguiente
filtro_clasificados['team']

0            Brazil
1            Mexico
4       Netherlands
5             Chile
8          Colombia
9            Greece
12       Costa Rica
13          Uruguay
16           France
17      Switzerland
20        Argentina
21          Nigeria
24          Germany
25    United States
28          Belgium
29          Algeria
Name: team, dtype: object

##### **g. ganar al menos un partido**

In [28]:
filtro_partidos_ganados = mundial_2014[mundial_2014['juegos_ganados'] >= 1]
filtro_partidos_ganados

team     continent group  group_pos  cantidad_juegos  \
0                   Brazil  southamerica     A          1                3   
1                   Mexico  northamerica     A          2                3   
2                  Croatia        europe     A          3                3   
4              Netherlands        europe     B          1                3   
5                    Chile  southamerica     B          2                3   
6                    Spain        europe     B          3                3   
8                 Colombia  southamerica     C          1                3   
9                   Greece        europe     C          2                3   
10             Ivory Coast        africa     C          3                3   
12              Costa Rica  northamerica     D          1                3   
13                 Uruguay  southamerica     D          2                3   
14                   Italy        europe     D          3                3   
16                  France        europe     E          1                3   
17             Switzerland        europe     E          2                3   
18                 Ecuador  southamerica     E          3                3   
20               Argentina  southamerica     F          1                3   
21                 Nigeria        africa     F          2                3   
22  Bosnia and Herzegovina        europe     F          3                3   
24                 Germany        europe     G          1                3   
25           United States  northamerica     G          2                3   
26                Portugal        europe     G          3                3   
28                 Belgium        europe     H          1                3   
29                 Algeria        africa     H          2                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0                2                 1                0            7   
1                2                 1                0            4   
2                1                 0                2            6   
4                3                 0                0           10   
5                2                 0                1            5   
6                1                 0                2            4   
8                3                 0                0            9   
9                1                 1                1            2   
10               1                 0                2            4   
12               2                 1                0            4   
13               2                 0                1            4   
14               1                 0                2            2   
16               2                 1                0            8   
17               2                 0                1            7   
18               1                 1                1            3   
20               3                 0                0            6   
21               1                 1                1            3   
22               1                 0                2            4   
24               2                 1                0            7   
25               1                 1                1            4   
26               1                 1                1            4   
28               3                 0                0            4   
29               1                 1                1            6   

    goles_contra goles_diferencia  puntos  clasificado  
0              2                5       7            1  
1              1                3       7            1  
2              6                0       3            0  
4              3                7       9            1  
5              3                2       6            1  
6              7               −3       3            0  
8              2                7       9            1  
9           

In [29]:
# Variable de cantidad de equipos que ganaron más de un partido
conteo_eq_que_ganaron = filtro_partidos_ganados['juegos_ganados'].value_counts()

conteo_eq_que_ganaron


juegos_ganados
1    11
2     8
3     4
Name: count, dtype: int64

In [30]:
# Variable de cantidad de partidos ganados, que es distinta, ya que 1 equipo pudo ganar hasta 3 partidos.

conteo_partidos_ganados = filtro_partidos_ganados['juegos_ganados'].sum()
conteo_partidos_ganados


np.int64(39)

In [32]:
# Visualización para distinguir ambas variables
total_eq = 0
for n_ganados, cantidad in conteo_eq_que_ganaron.items():
    if n_ganados == 1:
        print(f'Selecciones con {n_ganados} partido ganado: {cantidad}.')
    else:
        print(f'Selecciones con {n_ganados} partidos ganados: {cantidad}.')

    total_eq += cantidad
print(f"Total equipos que ganaron al menos 1 partido: {total_eq}") 
print(f"Total de partidos ganados: {conteo_partidos_ganados}") 


Selecciones con 1 partido ganado: 11.
Selecciones con 2 partidos ganados: 8.
Selecciones con 3 partidos ganados: 4.
Total equipos que ganaron al menos 1 partido: 23
Total de partidos ganados: 39


In [33]:
# Probabilidad de que una selección gane al menos 1 partido. 

p_almenos1partido = conteo_eq_que_ganaron.sum() / len(mundial_2014) 
print(f"La probabilidad de un equipo de ganar al menos 1 partido es de: {p_almenos1partido}")

La probabilidad de un equipo de ganar al menos 1 partido es de: 0.71875


##### **h. pertenecer a África y clasificar a segunda ronda**

La fórmula a utilizar en los siguientes ejercicios sería:
P (AyB) = P(A) x P (B | A)

In [36]:
# P de pertenecer a África
f'Probabilidad de pertenecer a África: {p_africa}'

'Probabilidad de pertenecer a África: 0.16'

In [37]:
# P de pertenecer a clasificar a segunda ronda
f'Probabilidad de pertenecer a segunda ronda: {p_clasificados}'

'Probabilidad de pertenecer a segunda ronda: 0.5'

In [38]:
# Probabilidad de pertenecer a África Y clasificar a segunda ronda   ---> P (A|B) =  P(A) * P (B|A) 

# Como ambos sucesos son independientes (no hay limitaciones para clasificar) ---> P (B|A) = P (B)
# 
#Entonces quedaría la fórmula --------->  P (A ∩ B) =  P(A) * P (B)

p_segunda_ronda_y_africa =  p_africa * p_clasificados

f'La probabilidad de que sea de África y pase a segunda ronda es de: {p_segunda_ronda_y_africa} o de un {p_segunda_ronda_y_africa * 100}%.'

'La probabilidad de que sea de África y pase a segunda ronda es de: 0.08 o de un 8.0%.'

##### **i. pertenecer a Europa y clasificar a segunda ronda**

In [39]:
f'Probabilidad de pertenecer a Europa: {p_europa}'

'Probabilidad de pertenecer a Europa: 0.41'

In [40]:
f'Probabilidad de pertenecer a segunda ronda: {p_clasificados}'

'Probabilidad de pertenecer a segunda ronda: 0.5'

In [41]:
p_segunda_ronda_y_europa =  p_europa * p_clasificados

f'La probabilidad de que sea de Europa y pase a segunda ronda es de: {p_segunda_ronda_y_europa} o de un {p_segunda_ronda_y_europa * 100}%.'

'La probabilidad de que sea de Europa y pase a segunda ronda es de: 0.205 o de un 20.5%.'

##### **j. pertenecer a Asia y clasificar a segunda ronda**

In [42]:
f'Probabilidad de pertenecer a Asia: {p_asia}'


'Probabilidad de pertenecer a Asia: 0.12'

In [43]:
f'Probabilidad de pertenecer a segunda ronda: {p_clasificados}'

'Probabilidad de pertenecer a segunda ronda: 0.5'

In [44]:
p_segunda_ronda_y_asia =  p_asia * p_clasificados

f'La probabilidad de que sea de Asia y pase a segunda ronda es de: {p_segunda_ronda_y_asia} o de un {p_segunda_ronda_y_asia * 100}%.'

'La probabilidad de que sea de Asia y pase a segunda ronda es de: 0.06 o de un 6.0%.'

##### **k. pertenecer a Sudamérica y clasificar a segunda ronda**

In [45]:
f'Probabilidad de pertenecer a Sudamérica: {p_sudamerica}'

'Probabilidad de pertenecer a Sudamérica: 0.19'

In [46]:
f'Probabilidad de pertenecer a segunda ronda: {p_clasificados}'

'Probabilidad de pertenecer a segunda ronda: 0.5'

In [47]:
p_segunda_ronda_y_sudamerica =  p_sudamerica * p_clasificados

f'La probabilidad de que sea de Sudamérica y pase a segunda ronda es de: {p_segunda_ronda_y_sudamerica} o de un {p_segunda_ronda_y_sudamerica * 100}%.'

'La probabilidad de que sea de Sudamérica y pase a segunda ronda es de: 0.095 o de un 9.5%.'

##### **l. pertenecer a Norteamérica y clasificar a segunda ronda**

In [48]:
f'Probabilidad de pertenecer a Norteamérica: {p_norteamerica}'

'Probabilidad de pertenecer a Norteamérica: 0.12'

In [49]:
p_segunda_ronda_y_norteamerica =  p_norteamerica * p_clasificados

f'La probabilidad de que sea de Norteamérica y pase a segunda ronda es de: {p_segunda_ronda_y_norteamerica} o de un {p_segunda_ronda_y_norteamerica * 100}%.'

'La probabilidad de que sea de Norteamérica y pase a segunda ronda es de: 0.06 o de un 6.0%.'

##### **m. pertenecer a África o clasificar a segunda ronda**

La fórmula a utilizar en los siguientes ejercicios sería:
P (A u B) = P(A) + P (B) - P (A ∩ B)

In [50]:
f'Probabilidad de pertenecer a África: {p_africa} + Probabilidad de clasificar a segunda ronda: {p_clasificados} = {p_africa + p_clasificados} '

'Probabilidad de pertenecer a África: 0.16 + Probabilidad de clasificar a segunda ronda: 0.5 = 0.66 '

In [51]:
# Unión
p_segunda_ronda_o_africa = (p_africa + p_clasificados) - p_segunda_ronda_y_africa
f'Probabilidad pertenecer a África o clasificar a segunda ronda: {round(p_segunda_ronda_o_africa, 4)}'

'Probabilidad pertenecer a África o clasificar a segunda ronda: 0.58'

##### **n. pertenecer a Europa o clasificar a segunda ronda**

In [52]:
f'Probabilidad de pertenecer a Europa: {p_europa} + Probabilidad de clasificar a segunda ronda: {p_clasificados} = {p_europa + p_clasificados} '

'Probabilidad de pertenecer a Europa: 0.41 + Probabilidad de clasificar a segunda ronda: 0.5 = 0.9099999999999999 '

In [53]:
p_segunda_ronda_o_europa = (p_europa + p_clasificados) - p_segunda_ronda_y_europa
f'Probabilidad pertenecer a Europa o clasificar a segunda ronda: {round(p_segunda_ronda_o_europa, 4)}'

'Probabilidad pertenecer a Europa o clasificar a segunda ronda: 0.705'

##### **o. pertenecer a Asia o clasificar a segunda ronda**

In [54]:
f'Probabilidad de pertenecer a Asia: {p_asia} + Probabilidad de clasificar a segunda ronda: {p_clasificados} = {p_asia + p_clasificados} '

'Probabilidad de pertenecer a Asia: 0.12 + Probabilidad de clasificar a segunda ronda: 0.5 = 0.62 '

In [55]:
p_segunda_ronda_o_asia = (p_asia + p_clasificados) - p_segunda_ronda_y_asia
f'Probabilidad pertenecer a Asia o clasificar a segunda ronda: {round(p_segunda_ronda_o_asia, 4)}'

'Probabilidad pertenecer a Asia o clasificar a segunda ronda: 0.56'

##### **p. pertenecer a Sudamérica o clasificar a segunda ronda**

In [56]:
f'Probabilidad de pertenecer a Sudamérica: {p_sudamerica} + Probabilidad de clasificar a segunda ronda: {p_clasificados} = {p_sudamerica + p_clasificados} '

'Probabilidad de pertenecer a Sudamérica: 0.19 + Probabilidad de clasificar a segunda ronda: 0.5 = 0.69 '

In [57]:
p_segunda_ronda_o_sudamerica = (p_sudamerica + p_clasificados) - p_segunda_ronda_y_sudamerica
f'Probabilidad pertenecer a Sudamérica o clasificar a segunda ronda: {round(p_segunda_ronda_o_sudamerica, 4)}'

'Probabilidad pertenecer a Sudamérica o clasificar a segunda ronda: 0.595'

##### **q. pertenecer a Norteamérica o clasificar a segunda ronda**

In [58]:
f'Probabilidad de pertenecer a Norteamérica: {p_norteamerica} + Probabilidad de clasificar a segunda ronda: {p_clasificados} = {p_norteamerica + p_clasificados} '

'Probabilidad de pertenecer a Norteamérica: 0.12 + Probabilidad de clasificar a segunda ronda: 0.5 = 0.62 '

In [59]:
p_segunda_ronda_o_norteamerica = (p_norteamerica + p_clasificados) - p_segunda_ronda_y_norteamerica
f'Probabilidad pertenecer a Norteamérica o clasificar a segunda ronda: {round(p_segunda_ronda_o_norteamerica, 4)}'

'Probabilidad pertenecer a Norteamérica o clasificar a segunda ronda: 0.56'

##### **r. pertenecer a África y haber ganado al menos un partido**

In [60]:
# Se calcula la probabilidad multiplicando las dos probabilidades sacadas anteriormente, dado que son sucesos independientes.


# Serían probabilidades teóricas, dado que no se basan en valores reales del dataframe
p_africa_y_ganar1partido = round((p_africa * p_almenos1partido),3)

print(f"La probabilidad de pertenecer a África y haber ganado al menos un partido es de: {p_africa_y_ganar1partido}")

La probabilidad de pertenecer a África y haber ganado al menos un partido es de: 0.115


In [61]:
# Probabilidad experimental, a partir de la intersección real del dataframe

cond_africa_y_ganar1partido =  mundial_2014[(mundial_2014["continent"]=="africa") & (mundial_2014["juegos_ganados"]>=1) ]
len(cond_africa_y_ganar1partido)

3

In [62]:
cond_africa_y_ganar1partido

team continent group  group_pos  cantidad_juegos  juegos_ganados  \
10  Ivory Coast    africa     C          3                3               1   
21      Nigeria    africa     F          2                3               1   
29      Algeria    africa     H          2                3               1   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
10                 0                2            4             5   
21                 1                1            3             3   
29                 1                1            6             5   

   goles_diferencia  puntos  clasificado  
10               −1       3            0  
21                0       4            1  
29                1       4            1

In [63]:
p_africa_y_ganar1partido_n =  len(cond_africa_y_ganar1partido) / len(mundial_2014) 

print(f"La probabilidad de pertenecer a África y haber ganado al menos un partido es de: {p_africa_y_ganar1partido_n}")

La probabilidad de pertenecer a África y haber ganado al menos un partido es de: 0.09375


En este caso, la probabilidad teórica sería más alta (0.115), dado que hay más selecciones africanas participando en relación a la cantidad de equipos de África que efectivamente ganaron al menos 1 partido (0.09375).

##### **s. pertenecer a Europa y haber ganado al menos un partido**

In [64]:
p_europa

0.41

In [65]:
# Probabilidad teórica
p_europa_y_ganar1partido = round((p_europa * p_almenos1partido),3)

print(f"La probabilidad teórica de pertenecer a Europa y haber ganado al menos un partido es de: {p_europa_y_ganar1partido}")

La probabilidad teórica de pertenecer a Europa y haber ganado al menos un partido es de: 0.295


In [66]:
# Probabilidad experimental, a partir de la intersección real del dataframe

cond_europa_y_ganar1partido =  mundial_2014[(mundial_2014["continent"]=="europe") & (mundial_2014["juegos_ganados"]>=1) ]

p_europa_y_ganar1partido_n =  len(cond_europa_y_ganar1partido) / len(mundial_2014) 

print(f"La probabilidad experimental de pertenecer a Europa y haber ganado al menos un partido es de: {p_europa_y_ganar1partido_n}")

La probabilidad experimental de pertenecer a Europa y haber ganado al menos un partido es de: 0.34375


##### t. pertenecer a Asia y haber ganado al menos un partido


In [71]:
# Probabilidad teórica

p_asia_y_ganar1partido = round((p_asia * p_almenos1partido),3)

print(f"La probabilidad de pertenecer a Asia ({p_asia}) y haber ganado al menos un partido ({p_almenos1partido}) es de: {p_asia_y_ganar1partido}")

La probabilidad de pertenecer a Asia (0.12) y haber ganado al menos un partido (0.71875) es de: 0.086


In [68]:
# Tantear las selecciones de Asia y juegos ganados (incluye 0 ganados, para verlos todos), en extenso
mundial_2014[(mundial_2014["continent"]=="asia") & (mundial_2014["juegos_ganados"]>=0) ]

team continent group  group_pos  cantidad_juegos  juegos_ganados  \
7     Australia      asia     B          4                3               0   
11        Japan      asia     C          4                3               0   
23         Iran      asia     F          4                3               0   
31  South Korea      asia     H          4                3               0   

    juegos_empatados  juegos_perdidos  goles_favor  goles_contra  \
7                  0                3            3             9   
11                 1                2            2             6   
23                 1                2            1             4   
31                 1                2            3             6   

   goles_diferencia  puntos  clasificado  
7                −6       0            0  
11               −4       1            0  
23               −3       1            0  
31               −3       1            0

In [72]:
# Probabilidad experimental, a partir de la intersección real del dataframe

cond_asia_y_ganar1partido =  mundial_2014[(mundial_2014["continent"]=="asia") & (mundial_2014["juegos_ganados"]>=1) ]
print(f'{len(cond_asia_y_ganar1partido)}')

p_asia_y_ganar1partido_n =  len(cond_asia_y_ganar1partido) / len(mundial_2014) 

print(f"La probabilidad de pertenecer a Asia y haber ganado al menos un partido es de: {p_asia_y_ganar1partido_n}")

0
La probabilidad de pertenecer a Asia y haber ganado al menos un partido es de: 0.0


##### u. pertenecer a Sudamérica y haber ganado al menos un partido

In [120]:
# Probabilidad teórica

p_sudamerica_y_ganar1partido = round((p_sudamerica * p_almenos1partido),3)

print(f"La probabilidad de pertenecer a Sudamérica y haber ganado al menos un partido es de: {p_sudamerica_y_ganar1partido}")

La probabilidad de pertenecer a Sudamérica y haber ganado al menos un partido es de: 0.137


In [121]:
# Probabilidad experimental, a partir de la intersección real del dataframe

cond_sudamerica_y_ganar1partido =  mundial_2014[(mundial_2014["continent"]=="southamerica") & (mundial_2014["juegos_ganados"]>=1) ]

p_sudamerica_y_ganar1partido_n =  len(cond_sudamerica_y_ganar1partido) / len(mundial_2014) 

print(f"La probabilidad de pertenecer a Sudamérica y haber ganado al menos un partido es de: {p_sudamerica_y_ganar1partido_n}")

La probabilidad de pertenecer a Sudamérica y haber ganado al menos un partido es de: 0.1875


##### v. pertenecer a Norteamérica y haber ganado al menos un partido

In [122]:
# Probabilidad teórica

p_norteamerica_y_ganar1partido = round((p_norteamerica * p_almenos1partido),3)
print(f"La probabilidad de pertenecer a Norteamérica y haber ganado al menos un partido es de: {p_norteamerica_y_ganar1partido}")

La probabilidad de pertenecer a Norteamérica y haber ganado al menos un partido es de: 0.086


In [123]:
# Probabilidad experimental, a partir de la intersección real del dataframe
cond_norteamerica_y_ganar1partido =  mundial_2014[(mundial_2014["continent"]=="northamerica") & (mundial_2014["juegos_ganados"]>=1) ]

p_norteamerica_y_ganar1partido_n =  len(cond_norteamerica_y_ganar1partido) / len(mundial_2014) 

print(f"La probabilidad de pertenecer a Norteamérica y haber ganado al menos un partido es de: {p_norteamerica_y_ganar1partido_n}")

La probabilidad de pertenecer a Norteamérica y haber ganado al menos un partido es de: 0.09375


##### w. haber ganado al menos un partido y clasificar a segunda ronda

In [143]:
print(f"Cantidad de equipos que han ganado desde 1 partido: {conteo_desde_1_partido_ganado}")

Cantidad de equipos que han ganado desde 1 partido: 23


In [134]:
# probabilidad de haber ganado un partido
p_almenos1partido

np.float64(0.71875)

In [135]:
# prob. clasificar a segunda ronda
p_clasificados

np.float64(0.5)

In [137]:
# Ambos son eventos independientes, dado que el haber ganado un partido no asegura clasificar

p_ganaralmenos1partido_y_clasificar = p_almenos1partido * p_clasificados

print(f"La probabilidad de haber ganado al menos un partido y clasificar a segunda ronda sería de:  {p_ganaralmenos1partido_y_clasificar}")



La probabilidad de haber ganado al menos un partido y clasificar a segunda ronda sería de:  0.359375


In [139]:
# Sólo por curiosidad, sumo para ver eventos favorables desde 2 partidos ganados, por equipo
conteo_desde_2_partidos_ganados = conteo_partidos_ganados[2] + conteo_partidos_ganados[3] 
print(f"Suma de equipos que han ganado desde 2 partidos: {conteo_desde_2_partidos_ganados}")

Suma de equipos que han ganado desde 2 partidos: 12


In [141]:
# Probabilidad de ganar al menos 2 partidos. 
p_almenos2partidos = conteo_desde_2_partidos_ganados / len(mundial_2014)
print(f"La probabilidad de un equipo de ganar al menos 2 partido es de: {p_almenos2partidos}")

La probabilidad de un equipo de ganar al menos 2 partido es de: 0.375


In [142]:
p_ganaralmenos2partidos_y_clasificar = p_almenos2partidos * p_clasificados
print(f"La probabilidad de haber ganado al menos dos partidos y clasificar a segunda ronda sería de:  {p_ganaralmenos2partidos_y_clasificar}")


La probabilidad de haber ganado al menos dos partidos y clasificar a segunda ronda sería de:  0.1875


In [ ]:
# Esta condición seŕia dependiente, por lo que la prob P (A y B)= P(A) x P(B | A)

p_ganaralmenos1partido_y_clasificar =   p_clasificados * p_almenos1partido / p_almenos1partido

print(f"La probabilidad de haber ganado al menos un partido y clasificar a segunda ronda sería de:  {p_ganaralmenos1partido_y_clasificar}")

La probabilidad de haber ganado al menos un partido y clasificar a segunda ronda sería de:  0.5


In [ ]:
# Esta condición seŕia dependiente, por lo que la prob P (A y B)= P(A) x P(B | A)
# P(B | A) = (P(A | B) * P(B)) / P(A)

p_ganaralmenos1partido_y_clasificar =   p_clasificados * p_almenos1partido / p_almenos1partido

print(f"La probabilidad de haber ganado al menos un partido y clasificar a segunda ronda sería de:  {p_ganaralmenos1partido_y_clasificar}")

-
-
-
___0___o___
-
-
-

#### 2. A partir de las probabilidades anteriores, y sin realizar conteos, calcular las siguientes probabilidades:


In [227]:
# P ( a | b ) = P ( a y b) / b

#### a. pertenecer a Europa, si se sabe que clasificó

In [127]:
# a =pertenece a Europa
# b = clasificó

print(f"La probabilidad es de: {p_europa_segunda_ronda / p_clasificados}")

La probabilidad es de: 0.38


#### b. pertenecer a Sudamérica, si se sabe que clasificó

In [124]:
# a =pertenece a Sudamérica
# b = clasificó

print(f"La probabilidad es de: {p_sudamerica_segunda_ronda / p_clasificados}")

La probabilidad es de: 0.32


#### c. pertenecer a África, si se sabe que ganó al menos un partido

In [128]:
# a =pertenece a África
# b = Ganó al menos 1 partido

print(f"La probabilidad es de: {p_africa_y_ganar1partido_n / p_almenos1partido}")


La probabilidad es de: 0.13043478260869565


#### d. haber ganado al menos un partido, si se sabe que clasificó

In [129]:
# a = Ganó al menos 1 partido
# b = Clasificó


print(f"La probabilidad es de: {p_ganaralmenos1partido_y_clasificar / p_clasificados}")



La probabilidad es de: 1.0


#### e. haber ganado al menos un partido, si se sabe que pertenece a Norteamérica

In [130]:
# a = Ganó al menos 1 partido
# b = Pertnece a Norteamérica

print(f"La probabilidad es de: {p_norteamerica_y_ganar1partido_n / p_norteamerica}")


La probabilidad es de: 0.78125


#### f. NO haber clasificado, si se sabe que pertenece a África

In [144]:
# a = No clasificó
# b = Pertenece a África

# Para saber los no clasificados, se saca el complemento
print(f"Probabilidad de clasificar a 2da fase: {p_clasificados}")
print(f"Probabilidad de África a 2da fase: {p_africa}")

complemento_no_clasificados = 1 - p_africa


print(f"La probabilidad es de: {(complemento_no_clasificados * p_africa) / p_africa}")

Probabilidad de clasificar a 2da fase: 0.5
Probabilidad de África a 2da fase: 0.16
La probabilidad es de: 0.84


-
-
-
___0_FiN__o___
-
-
-

### Requerimientos
Dentro del archivo de Jupyter Notebook debes ir ejecutando las siguientes acciones y explicar lo que estás haciendo. Debes considerar los siguientes aspectos.

1. Define condiciones para calcular probabilidades con Python. (2 Puntos)

2. Calcula probabilidades de sucesos con datos en DataFrames. (3 Puntos)

3. Interpreta y aplica la probabilidad condicional y el teorema de Bayes. (5 Puntos)